pytorch_lightning/metrics/classification/utils.py

```diff
@@ -0,0 +1,390 @@
+        num_classes: number of classes
+        is_multiclass: if True, treat binary and multi-label inputs as multi-class or multi-dim
+            multi-class with 2 classes, respectively. If False, treat multi-class and multi-dim
+            multi-class inputs with 1 or 2 classes as binary and multi-label, respectively.
+            Defaults to None, which treats inputs as they appear.
+    """
+
+    if target.is_floating_point():
+        raise ValueError("target has to be an integer tensor")
+    elif target.min() < 0:
+        raise ValueError("target has to be a non-negative tensor")
+
+    preds_float = preds.is_floating_point()
+    if not preds_float and preds.min() < 0:
+        raise ValueError("if preds are integers, they have to be non-negative")
+
+    if not preds.shape[0] == target.shape[0]:
+        raise ValueError("preds and target should have the same first dimension.")
+
+    if preds_float:
+        if preds.min() < 0 or preds.max() > 1:
+            raise ValueError("preds should be probabilities, but values were detected outside of [0,1] range")
+
+    if threshold > 1 or threshold < 0:
+        raise ValueError("Threshold should be a probability in [0,1]")
+
+    if is_multiclass is False and target.max() > 1:
+        raise ValueError("If you set is_multiclass=False, then target should not exceed 1.")
+
+    if is_multiclass is False and not preds_float and preds.max() > 1:
+        raise ValueError("If you set is_multiclass=False and preds are integers, then preds should not exceed 1.")
+
+    # Check that shape/types fall into one of the cases
+    if preds.ndim == target.ndim:
+        if preds.shape != target.shape:
+            raise ValueError(
+                "preds and targets should have the same shape",
+                f" got preds shape = {preds.shape} and target shape = {target.shape}.",
+            )
+        if preds_float and target.max() > 1:
+            raise ValueError("if preds and target are of shape (N, ...) and preds are floats, target should be binary")
+
+        # Get the case
+        if preds.ndim == 1 and preds_float:
+            case = "binary"
+        elif preds.ndim == 1 and not preds_float:
+            case = "multi-class"
+        elif preds.ndim > 1 and preds_float:
+            case = "multi-label"
+        else:
+            case = "multi-dim multi-class"
+
+        implied_classes = torch.prod(torch.Tensor(list(preds.shape[1:])))
+
+    elif preds.ndim == target.ndim + 1:
+        if not preds_float:
+            raise ValueError("if preds have one dimension more than target, preds should be a float tensor")
+        if not preds.shape[:-1] == target.shape:
+            if preds.shape[2:] != target.shape[1:]:
+                raise ValueError(
+                    "if preds if preds have one dimension more than target, the shape of preds should be"
+                    "either of shape (N, C, ...) or (N, ..., C), and of targets of shape (N, ...)"
```

Comment:
```suggestion
                    " either of shape (N, C, ...) or (N, ..., C), and of targets of shape (N, ...)"
```
if multiline, the next line starts with space otherwise there is no space between the last word on the previous line and fist word on new line...

Reply:
Done